test/unit-test/mqtt_agent_utest.c

```diff
@@ -81,14 +136,307 @@ int suiteTearDown( int numFailures )
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    /* Test if NULL is passed for any of the function pointers. */
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgCtx, &networkBuffer, &transportInterface, NULL, incomingCallback, incomingPacketContext );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgCtx, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    msgCtx.pMsgCtx = &msg;
+
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgCtx, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    msgCtx.send = send;
+
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgCtx, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    msgCtx.recv = receive;
+
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgCtx, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    msgCtx.releaseCommand = commandRelease;
+
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgCtx, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+}
+
+/**
+ * @brief Test that any NULL parameter causes MQTTAgent_ResumeSession to return MQTTIllegalState.
+ */
+void test_MQTTAgent_ResumeSession_Invalid_Params( void )
+{
+    MQTTAgentContext_t mqttAgentContext;
+    bool sessionPresent = true;
+    MQTTStatus_t mqttStatus;
+
+    AgentMessageInterface_t msgCtx;
+    MQTTFixedBuffer_t networkBuffer;
+    TransportInterface_t transportInterface;
+    IncomingPublishCallback_t incomingCallback;
+    void * incomingPacketContext;
+    AgentMessageContext_t msg;
+
+
+
+    msgCtx.pMsgCtx = &msg;
+    msgCtx.send = send;
+    msgCtx.recv = receive;
+    msgCtx.releaseCommand = commandRelease;
+    msgCtx.getCommand = getCommand;
+    MQTT_Init_ExpectAnyArgsAndReturn( MQTTSuccess );
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgCtx, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+
+
+    /* Check that MQTTIllegalState is returned if any NULL mqttAgentContext is passed. */
+    mqttStatus = MQTTAgent_ResumeSession( NULL, sessionPresent );
+    TEST_ASSERT_EQUAL( MQTTIllegalState, mqttStatus );
+
+    /* Check that MQTTIllegalState is returned if packet id is 0 */
+    mqttAgentContext.mqttContext.nextPacketId = 0;
+    mqttStatus = MQTTAgent_ResumeSession( &mqttAgentContext, sessionPresent );
+    TEST_ASSERT_EQUAL( MQTTIllegalState, mqttStatus );
+}
+
+
+void test_MQTTAgent_ResumeSession_session_present_with_packetId_Invalid( void )
```

Comment:
A better name for this would be `test_MQTTAgent_ResumeSession_no_resent_publishes`, since that's what a 0 return from `MQTT_PublishToResend` indicates. However, I think it would make more sense to combine this function with `test_MQTTAgent_ResumeSession_publish_resend_success` and `test_MQTTAgent_ResumeSession_ack_packetId_not_match` since they all test success cases dealing with various packet IDs.